.github/workflows/build.yml

```diff
@@ -270,6 +279,42 @@ jobs:
           rm -rf temp
           mv signed temp
 
+      - name: Generate VSA
+        if: env.shouldSign == 'true'
+        env:
+          AWS_KMS_REGION:  ${{ job.build.steps.auth.with.args.aws-region }}
+          AWS_KMS_KEY_ARN: ${{ job.build.steps.sign.env.AWS_KMS_KEY_ARN }}
+        run: |
+          args+=(
+            --mount "type=bind,src=$PWD/signed,dst=/doi-build/signed"
+
+            # https://explore.ggcr.dev/?repo=docker/image-signer-verifier
+            "docker/image-signer-verifier:$BASHBREW_IMAGE_SIGNER"
+
+            verify
```

Comment:
This seems like a strange name for a command that's apparently doing some actual signing of something? :grimacing:

Is there a short version of what each of these signatures/objects are and what they represent?  I realize this is still experimental, so I'm not expecting _super_ fleshed out documentation, but at the very least something we can use to explain to curious users who find them and ask about it would be really useful (not the least because I _myself_ don't understand them, so don't really feel good about publishing them until I do :innocent:).

Reply:
I also agree that having VSA generation be overloaded in the `verify` command is confusing. We had it as an arg to `sign` (`--vsa`) and realized it didn't make sense to create a verification summary (vsa) without verifying a policy. That doesn't mean it has to be overloaded in the `verify` command. I would prefer a separate command (e.g. `vsa` or `attest --vsa`) so that is less confusing. It also pulls all of the signing args out of `verify` and would give good separation.

> Is there a short version of what each of these signatures/objects are and what they represent?

I'll take a pass and suggest some comments to add

Reply:
We had a really nice offline session where Joel explained the workflow in more detail -- the short version is that the `sign` command signs all the attachment objects (all SBOMs + the BuildKit provenance doc) and pushes those signed objects as individual "signed attestation" artifact manifests to the specified "referrers" repository.

_This_ command then pulls a policy that lives in that `docker/tuf` repo (which is a Docker Hub repository that contains a TUF repository), verifies that the image passes the policy (hence the `verify` command here), and generates [a signed SLSA "VSA"](https://slsa.dev/spec/v1.0/verification_summary) (so there's some slight weirdness in the VSA because we're not fully following the rest of the SLSA spec in our formats/process but still trying to use their data structure) which is then _also_ pushed to the designated "referrers" repository.